spec/controllers/two_factor_authentication/otp_verification_controller_spec.rb

```diff
@@ -697,15 +697,38 @@
               in_account_creation_flow: false,
             }
 
+            response
+
             expect(@analytics).to have_received(:track_event).
               with('Multi-Factor Authentication Setup', properties)
 
-            expect(subject).to have_received(:create_user_event).with(:phone_confirmed)
-            expect(subject).to have_received(:create_user_event).exactly(:once)
+            expect(controller).to have_received(:create_user_event).with(:phone_confirmed)
+            expect(controller).to have_received(:create_user_event).exactly(:once)
+          end
+
+          it 'annotates with passed 2fa and resets a recaptcha assessment' do
+            assessment_id = 'projects/project-id/assessments/assessment-id'
+
+            controller.user_session[:phone_recaptcha_assessment_id] = assessment_id
+
+            annotator = instance_double(RecaptchaAnnotator)
+            expect(annotator).to receive(:annotate).once.with(
+              reason: RecaptchaAnnotator::AnnotationReasons::PASSED_TWO_FACTOR,
+            )
+
+            allow(RecaptchaAnnotator).to receive(:new).
+              with(assessment_id:, analytics: @analytics).
+              and_return(annotator)
```

Comment:
I don't like stubbing `new`, what if we added a class method `.annotate()` that wraps `.new().annotate()`?

Reply:
I might be more okay with class method implementation based on refactoring stemming from https://github.com/18F/identity-idp/pull/10522#discussion_r1589597913, if we don't need to have `RecaptchaAnnotator` deal with analytics, I think it could make more sense to have a class method `annotate(assessment_id:, reason:, assessment:)`.

Reply:
This was addressed indirectly in dcc31d13d9.